src/Stryker.Core/Stryker.Core.UnitTest/Reporters/JsonReporterTests.cs

```diff
@@ -18,184 +22,229 @@ namespace Stryker.Core.UnitTest.Reporters
+        }
+
+        [Fact]
+        public void JsonReport_WithMutationScoreUnderThresholdBreakHasDangerHealth()
+        {
+            var folderComponent = CreateProjectWith();
+
+            var report = JsonReport.Build(new StrykerOptions(thresholdHigh: 80, thresholdLow: 70, thresholdBreak: 67), folderComponent);
+
+            report.Files.First().Value.Health.ShouldBe(Health.Danger);
+        }
+
+        [Fact]
+        public void JsonReport_WithMutationScoreEqualToThresholdBreakHasDangerHealth()
+        {
+            var folderComponent = CreateProjectWith();
+
+            var report = JsonReport.Build(new StrykerOptions(thresholdHigh: 80, thresholdLow: 70, thresholdBreak: 66), folderComponent);
+
+            report.Files.First().Value.Health.ShouldBe(Health.Danger);
+        }
+
+        [Fact]
+        public void JsonReport_BuildReportReturnsSingletonJsonReport()
+        {
+            var folderComponent = CreateProjectWith();
+            var options = new StrykerOptions();
+
+            var firstReport = JsonReport.Build(options, folderComponent);
+            var secondReport = JsonReport.Build(options, folderComponent);
+
+            secondReport.ShouldBe(firstReport);
+        }
+
+        [Fact]
+        public void JsonReporter_OnAllMutantsTestedShouldWriteJsonToFile()
+        {
             var mockFileSystem = new MockFileSystem();
             var options = new StrykerOptions(thresholdBreak: 0, thresholdHigh: 80, thresholdLow: 60);
             var reporter = new JsonReporter(options, mockFileSystem);
 
-            reporter.OnAllMutantsTested(folder);
-            var reportPath = Path.Combine(options.BasePath, "StrykerOutput", "reports", $"mutation-report-{DateTime.Today.ToString("yyyy-MM-dd")}.json");
+            reporter.OnAllMutantsTested(CreateProjectWith());
+            var reportPath = Path.Combine(options.OutputPath, "reports", $"mutation-report.json");
             mockFileSystem.FileExists(reportPath).ShouldBeTrue($"Path {reportPath} should exist but it does not.");
-
-            var reportObject = JsonConvert.DeserializeObject<JsonReporter.JsonReportComponent>(mockFileSystem.GetFile(reportPath).TextContents);
-            reportObject.ThresholdHigh.ShouldBe(80);
-            reportObject.ThresholdLow.ShouldBe(60);
-            reportObject.ThresholdBreak.ShouldBe(0);
-
-            ValidateJsonReportComponent(reportObject, folder, "Warning");
-            ValidateJsonReportComponent(reportObject.ChildResults.ElementAt(0), folder.Children.ElementAt(0), "Good", 1);
-            ValidateJsonReportComponent(reportObject.ChildResults.ElementAt(1), folder.Children.ElementAt(1), "Danger", 1);
         }
 
-        private void ValidateJsonReportComponent(JsonReporter.JsonReportComponent jsonComponent, IReadOnlyInputComponent inputComponent, string health, int mutants = 0)
+        private IReadOnlyInputComponent CreateProjectWith(bool duplicateMutant = false, int mutationScore = 60)
```

Comment:
There already is a method just like this in HtmlReporterTests. I'd like to see this method shared across both tests.